web/src/hash_util.ts

```diff
@@ -193,6 +193,37 @@ export function parse_narrow(hash: string): Term[] | undefined {
     return terms;
 }
 
+/**
+ *
+ * @param url_str string containing the url
+ * @returns [true, [stream_name]] if the url is a valid stream url
+ * @returns [true, [stream_name, topic]] if the url is a valid stream topic url
+ * @returns [false, null] if the url is invalid
```

Comment:
I think returning `null` or an object containing `stream_name` and `topic` fields might be cleaner, rather than this highly variable tuple type?

Reply:
Sure